client/lib/shared.coffee

```diff
@@ -52,14 +52,28 @@ Houston._nested_field_lookup = (object, path) ->
 Houston._convert_to_correct_type = (field, val, collection) ->
   find_obj = {}
   find_obj[field] = $exists: true
-  sample_val = Houston._nested_field_lookup(collection.findOne(find_obj), field)
-  constructor = sample_val.constructor
-  if typeof sample_val == 'object'
-    new constructor(val)
-  else if typeof sample_val == 'boolean'
-    val == 'true'
+  one = collection.findOne(find_obj)
+  fields = _.object(_.map(Houston._collections.collections.findOne({name: collection.name}).fields, (field) -> # TODO it may be too heavy
+    [
+      field.name
+      type: field.type
+    ]
+  ))
+  constructor = window[fields[field].type]
+  if not one and constructor
+    if constructor is Boolean # special case for new Boolean()
+      val == 'true'
+    else
+      new constructor(val)
   else
-    constructor(val)
+    sample_val = Houston._nested_field_lookup(one, field)
+    constructor = sample_val.constructor
+    if typeof sample_val == 'object'
+      new constructor(val)
+    else if typeof sample_val == 'boolean'
+      val == 'true'
+    else
+      constructor(val)
```

Comment:
What this method used to do (and doing still without the schema enabled) is selecting _one_ document from the collection with field that we need to convert to 'correct' type (btw I strongly against word 'correct' in method naming, different things can be correct in different situations) and ejects it's constructor from value instance. Then it use this constructor on string which comes from UI. This method used in creating new documents from UI and editing field in existing documents, converting input string values to necessary type. Sincerely I don't like all this sampling stuff, it's unreliable in means Mongo collection [shouldn't but still] can have documents with different structures. Ideal solution _for me_ would be just use schema everywhere instead of sampling but it'll break Houston backward compartability. 

What it does if schema _is_ present (and I deduce it _hacky_ way as simpleSchema() on collection isn't accessible on client side) and sample document _isn't_ present is taking a type (as a string) from Huston collection 'collections' and trying to find it in global space (as String, Date, Number are globals) then presumes that this function found is a constructor. I don't like this solution but it is what I come with so far.